### PR TITLE
New improvements and fixes to pytest req updates plugin

### PR DESCRIPTION
**Pull request by jyejare**
### Problem Statement
- UV package manager support was not available
- Broker requirement not being handled well as optional backends/installations were activated
- pyYMAL requirement was not handled well
- airgun and nailgun git installations were not handled well as they don't have versions to compare 

### Solution
- UV package manager support is added
- Broker handled for optional backends/installations 
- pyYMAL requirement now handled
- airgun and nailgun git installations are handled and now wont be reported as deviation but would be updated 

Logs:
```
# pytest -s --collect-only tests/foreman/api/test_product.py --update-all-reqs
============= test session starts ===============
platform darwin -- Python 3.12.2, pytest-8.3.3, pluggy-1.5.0
Using Python 3.12.2 environment at /Users/jyejare/venv/uv_robo_master
 Updated https://github.com/SatelliteQE/airgun.git (98c53e4)
 Updated https://github.com/SatelliteQE/nailgun.git (08bbd9a)                                                                                                                                                         Resolved 83 packages in 2.83s
Audited 83 packages in 0.15ms
Required requirements are updated.
Optional Requirements are up to date.
To update requirements, run the pytest with '--update-required-reqs' OR '--update-all-reqs' option.

```

### CODE REVIEW

File: pytest_plugins/requirements/req_updater.py [hunk 1]
```diff
@@ -56,16 +64,60 @@ def opt_deviation(self):
...
-            )
+            if (
+                subprocess.run(
+                    f"{self.packagae_manager} install {' '.join(f"'{req}'" for req in self.req_deviation)}",
+                    shell=True,
+                    stdout=subprocess.PIPE,
+                ).returncode
+                == 0
+            ):
+                print('Required requirements are installed to be up-to-date.')
```
Comment (JacobCallahan) [inline pytest_plugins/requirements/req_updater.py:106]:
```suggestion
                print('Required requirements updated.')
```

File: pytest_plugins/requirements/req_updater.py [hunk 2]
```diff
@@ -56,16 +64,60 @@ def opt_deviation(self):
...
-            )
+            if (
+                subprocess.run(
+                    f"{self.packagae_manager} install {' '.join(f'\'{req}\'' for req in self.opt_deviation)}",
+                    shell=True,
+                    stdout=subprocess.PIPE,
+                ).returncode
+                == 0
+            ):
+                print('Optional requirements are installed to be up-to-date.')
```
Comment (JacobCallahan) [inline pytest_plugins/requirements/req_updater.py:121]:
```suggestion
                print('Optional requirements updated.')
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
minor phrasing. ack on the concept though

**Comment by jyejare** [comment]
@JacobCallahan Please revisit, all comments are fixed.

**Comment by jyejare** [comment]
@JacobCallahan Please revisit, all comments are fixed.

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**